ui/src/app/views/not-found/component.ts

```diff
@@ -0,0 +1,42 @@
+import {Location} from '@angular/common';
+import {Component, Injectable} from '@angular/core';
+import {NavigationError, Router} from '@angular/router';
+
+// Mostly borrowed from the Job Manager.
+@Component({
+    selector: 'app-not-found',
+    templateUrl: './component.html',
+    styleUrls: ['./component.css']
+})
+@Injectable()
+export class NotFoundComponent {
```

Comment:
May want to name more generically. This is listening on any failed initial navigation regardless of the status code.